### PR TITLE
Feature/more docs and index

### PR DESCRIPTION
**Pull request by gsaluja9**

### CODE REVIEW

File: docs/_static/parallelizer.dot [hunk 1]
```diff
@@ -0,0 +1,19 @@
+digraph parallelizer {
```
Comment (gsaluja9) [inline docs/_static/parallelizer.dot:1]: This is bases on a DSL : https://graphviz.org/doc/info/lang.html
Can be used to describe arbitrary graphs, that can be rendered.

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.8.10'
+    - uses: pre-commit/action@v2.0.3
```
Comment (luisremis) [inline .github/workflows/pre-commit.yml:16]:
```suggestion
    - uses: pre-commit/action@v2.0.3

```

File: aperturedb/Connector.py [hunk 3]
```diff
@@ -43,6 +43,22 @@
...
+    It lets the client execute any query as per the `Native Query specs <https://docs.aperturedata.io/>`_
+
+    Args:
+        str (host): Address of the host to connect to.
+        int (port): Port to connect to.
+        str (user): Username to specify while establishing a connection.
+        str (password): Password to specify while connecting to the db.
+        str (token): Token to use while connecting to the database.
+        object (session):
+        bool (use_ssl): Use SSL to encrypt communication with the database.
```
Comment (luisremis) [inline aperturedb/Connector.py:60]: is there a way to show the default values? In the case of use_ssl=True, for instance.

File: docs/_static/parallelizer.dot (hunk 1 above)
Comment (luisremis) [reply docs/_static/parallelizer.dot:1]: This is pretty cool, but you have to create it by hand? is there a way to automatically generate them? ()

File: docs/index.rst [hunk 4]
```diff
@@ -3,8 +3,38 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to aperturedb-python's documentation!
-=============================================
+Aperturedb-python's documentation.
```
Comment (luisremis) [inline docs/index.rst:6]:
```suggestion
ApertureDB Python Package Documentation.
```

**Review by luisremis — APPROVED** [review]
Looks good, thanks! take a look at minor suggestion as commits.

File: docs/_static/parallelizer.dot (hunk 1 above)
Comment (gsaluja9) [reply docs/_static/parallelizer.dot:1]: Will need to check something like that. Sounds quite radical, but I am not sure if there is something like that (cursory search)

File: aperturedb/Connector.py (hunk 3 above)
Comment (luisremis) [reply aperturedb/Connector.py:60]: oh I see, great, this is good enough, sorry I missed it.

File: docs/_static/parallelizer.dot (hunk 1 above)
Comment (luisremis) [reply docs/_static/parallelizer.dot:1]: mmm cool, I guess we would have to add stuff to the docstring for any tool to autogenerate something like that. Probably not worth it at this point. But we can leave the ones you created for reference and over time add more and experiment. If we realize they go out of sync fast and hard to maintain, we just remove them all.

**Review by luisremis — APPROVED** [review]

**Pull request merged**